packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp

```diff
@@ -0,0 +1,90 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for clientFormat decorator on duration.")
+@supportedBy("dpg")
+@scenarioService("/azure/client-generator-core/client-format/duration")
+namespace _Specs_.Azure.ClientGenerator.Core.ClientFormat.Duration;
+
+@global.Azure.ClientGenerator.Core.operationGroup
+@route("/query")
+namespace Query {
+  @route("/iso8601")
+  @scenario
+  @scenarioDoc("""
+  Test that default client format for a duration parameter is iso8601.
+  Expected query parameter `input=P40D`
+  """)
+  op iso8601(
+    @query
+    input: duration
+  ): NoContentResponse;
+
+  @route("/seconds")
+  @scenario
+  @scenarioDoc("""
+  Test seconds client format for a duration parameter.
+  Expected query parameter `input=35.6`
+  """)
+  op seconds(
+    @query
+    @global.Azure.ClientGenerator.Core.clientFormat("seconds")
+    input: duration
+  ): NoContentResponse;
+}
+
+@global.Azure.ClientGenerator.Core.operationGroup
+@route("/property")
+namespace Property {
+  model ISO8601DurationProperty {
+    value: duration;
+  }
+
+  model SecondsDurationProperty {
+    @global.Azure.ClientGenerator.Core.clientFormat("seconds")
```

Comment:
didn't we agree that we need 2 parameters here?  `units` and `format`.  I think there are two real examples both wanting duration in seconds as the units but one wanting a double and one wanting an int?

Reply:
or maybe have the format be `"seconds-int"`, `"seconds-float64"`

Reply:
That could work @timotheeguerin, typically would lean towards 2 params here but the combined single param is less work we can do that.

@iscai-msft I would lean towards tcgc parsing / splitting this into two parameters that get passed to emitters.

Reply:
@m-nash @joheredi is working on that right now, we do have 2 options here: we can wait for him to be done, or we can go ahead and release the test now. the more i think though i think you're right, we should wait until all of the support is into tcgc

Reply:
@iscai-msft after the tcgc update, I could help to update the test then.